runtime/compiler/runtime/JITClientSession.cpp

```diff
@@ -195,21 +195,19 @@ ClientSessionData::processUnloadedClasses(const std::vector<TR_OpaqueClassBlock*
          // here because compilation object hasn't been initialized yet
 
          std::string sigStr(sigLen, 0);
-         if (className[0] == '[')
-            {
-            memcpy(&sigStr[0], className, sigLen);
-            }
-         else
-            {
+         if (className[0] == '['){
+            sigStr = className;
+        }
+         else{
```

Comment:
`className` may not be NUL-terminated so the changes here and on line 208 are not appropriate.